pgmpy/metrics/metrics.py

```diff
@@ -465,3 +465,82 @@ def SHD(true_model, est_model):
     shd = shd + (np.sum((d + d.T) > 0) / 2)
 
     return int(shd)
+
+
+def self_compatibility_score(
+    estimator_class,
+    data: pd.DataFrame,
+    num_resamples: int = 20,
+    with_replacement: bool = True,
+    random_state: int = None,
+) -> float:
+    """
+    Computes the self-compatibility of a structure learner class.
```

Comment:
I would suggest describing it in much more detail. Its not a very standard method, so users will need to get an idea of what result they will get out of the method.